runtimes/node-18.0/server.js

```diff
@@ -30,13 +30,36 @@ const server = micro(async (req, res) => {
         headers[header.toLowerCase()] = req.headers[header];
     });
 
+    const scheme = (req.headers['x-forwarded-proto'] ?? 'http');
+    const host = req.headers['host'].includes(':') ? req.headers['host'].split(':')[0] : req.headers['host'];
+    const port = +(req.headers['host'].includes(':') ? req.headers['host'].split(':')[1] : '80');
+    const path = req.url.includes('?') ? req.url.split('?')[0] : req.url;
+    const queryString = req.url.includes('?') ? req.url.split('?')[1] : '';
+    const query = {};
+    for(const param of queryString.split('&')) {
+        let [key, ...valueArr] = param.split('=');
+        const value = valueArr.join('=');
+
+        if(key) {
+            query[key] = value ?? '';
+        }
+    }
+
+    const url = `${scheme}://${host}${port === 80 ? '' : `:${port}`}${path}${queryString === '' ? '' : `?${queryString}`}`;
```

Comment:
If scheme is `https` and port `443` we should not include the port

Reply:
Fixed, updated tests.